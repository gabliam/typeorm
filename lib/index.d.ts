import { Gabliam } from '@gabliam/core';
export * from "typeorm";
export default function (framework: Gabliam): Promise<void>;
