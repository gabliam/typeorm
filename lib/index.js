"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : new P(function (resolve) { resolve(result.value); }).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
function __export(m) {
    for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
}
const typeorm_1 = require("typeorm");
__export(require("typeorm"));
function default_1(framework) {
    return __awaiter(this, void 0, void 0, function* () {
        let driver = framework.container.get('DriverOptions');
        const conn = yield typeorm_1.createConnection({
            driver,
            entities: [
                `${framework.discoverPath}/**/*{.js,.ts}`
            ],
            autoSchemaSync: true,
        });
        framework
            .container
            .bind(typeorm_1.Connection)
            .toConstantValue(conn);
    });
}
Object.defineProperty(exports, "__esModule", { value: true });
exports.default = default_1;
;
