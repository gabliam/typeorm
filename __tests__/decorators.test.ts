import { RegistryMetada } from '@gabliam/core/lib/interfaces';
import {
  Entity,
  AbstractEntity,
  ClassEntityChild,
  ClosureEntity,
  EmbeddableEntity,
  SingleEntityChild
} from '../src/decorators';
import { METADATA_KEY as CORE_METADATA_KEY } from '@gabliam/core/lib/constants';

[
  Entity,
  AbstractEntity,
  ClassEntityChild,
  ClosureEntity,
  EmbeddableEntity,
  SingleEntityChild
].forEach(deco => {
  test(`${deco.name} decorators`, () => {
    @deco()
    class TestEntity {}

    const entityMetadata: RegistryMetada[] = Reflect.getMetadata(
      CORE_METADATA_KEY.register,
      TestEntity
    );

    expect(entityMetadata).toMatchSnapshot();
  });
});
