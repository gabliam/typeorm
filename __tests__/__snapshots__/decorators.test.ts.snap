// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AbstractEntity decorators 1`] = `
Object {
  "type": "EntityTypeOrm",
  "value": Object {
    "id": [Function],
    "target": [Function],
  },
}
`;

exports[`ClassEntityChild decorators 1`] = `
Object {
  "type": "EntityTypeOrm",
  "value": Object {
    "id": [Function],
    "target": [Function],
  },
}
`;

exports[`ClosureEntity decorators 1`] = `
Object {
  "type": "EntityTypeOrm",
  "value": Object {
    "id": [Function],
    "target": [Function],
  },
}
`;

exports[`EmbeddableEntity decorators 1`] = `
Object {
  "type": "EntityTypeOrm",
  "value": Object {
    "id": [Function],
    "target": [Function],
  },
}
`;

exports[`Entity decorators 1`] = `
Object {
  "type": "EntityTypeOrm",
  "value": Object {
    "id": [Function],
    "target": [Function],
  },
}
`;

exports[`SingleEntityChild decorators 1`] = `
Object {
  "type": "EntityTypeOrm",
  "value": Object {
    "id": [Function],
    "target": [Function],
  },
}
`;
